### ISSUE TITLE
Exception is thrown from MakeNonCacheable when applying caching to a 404 response

### ISSUE THREAD

**Issue by jacojvv-dev** [aliostad/CacheCow#235]
When returning a 404 (return NotFound()) from an API controller a null reference exception is raised from within the MakeNonCacheable method.

Upon further inspection it seems to be related to the addition of the expires header on the headers of the response.

```csharp
//
// Summary:
//     /// Makes a response non-cacheable by all the means available to mankind including
//     nuclear ///
//
// Parameters:
//   response:
public static void MakeNonCacheable(this HttpResponseMessage response)
{
    //IL_0011: Unknown result type (might be due to invalid IL or missing references)
    //IL_001b: Expected O, but got Unknown
    //IL_0022: Unknown result type (might be due to invalid IL or missing references)
    //IL_0027: Unknown result type (might be due to invalid IL or missing references)
    //IL_002f: Unknown result type (might be due to invalid IL or missing references)
    //IL_003c: Expected O, but got Unknown
    response.get_Headers().get_Pragma().Add(new NameValueHeaderValue("no-cache"));
    HttpResponseHeaders headers = response.get_Headers();
    CacheControlHeaderValue val = new CacheControlHeaderValue();
    val.set_NoCache(true);
    val.set_NoStore(true);
    headers.set_CacheControl(val);
    response.get_Content().get_Headers().set_Expires((DateTimeOffset?)DateTimeOffset.Now.AddDays(-1.0));
}
```

Specifically, the following line:

```csharp
response.get_Content().get_Headers().set_Expires((DateTimeOffset?)DateTimeOffset.Now.AddDays(-1.0));
```

Where Response.Content is null.

The only way to currently bypass this exception is by adding an empty body or specifying ApplyNoCacheNoStoreForNonCacheableResponse = true on the attribute (which would cause 404s to be cached).

Is this something that needs to be fixed or is this behavior intended?

**Comment by aliostad** [comment]
Thanks for raising. This is a bug.

**Comment by aliostad** [comment]
Thanks for reporting, I am surprised it was not found before. In any case fix is there with version 2.5.1.

Cheers
Ali
